api/wasm/cpp/proxy_wasm_intrinsics.proto

```diff
@@ -1,7 +1,82 @@
 syntax = 'proto3';
 
+import "google/protobuf/any.proto";
 import "google/protobuf/struct.proto";
 
 message WKT {
   google.protobuf.Struct struct = 1;
 }
+
+message DataSource {
+  oneof specifier {
+    string filename = 1;
+    bytes inline_bytes = 2;
+    string inline_string = 3;
+  }
+}
+
+message GrpcService {
+  message EnvoyGrpc {
+    string cluster_name = 1;
+  }
+  message GoogleGrpc {
+    string target_uri = 1;
+    message SslCredentials {
+      DataSource root_certs = 1;
+      DataSource private_key = 2;
+      DataSource cert_chain = 3;
+    }
+    message GoogleLocalCredentials {
+    }
+    message Empty {
+    }
+    message ChannelCredentials {
+      oneof credential_specifier {
+        SslCredentials ssl_credentials = 1;
+        Empty google_default = 2;
+        GoogleLocalCredentials local_credentials = 3;
+      }
+    }
+    ChannelCredentials channel_credentials = 2;
+    message CallCredentials {
+      message ServiceAccountJWTAccessCredentials {
+        string json_key = 1;
+        uint64 token_lifetime_seconds = 2;
+      }
+      message GoogleIAMCredentials {
+        string authorization_token = 1;
+        string authority_selector = 2;
+      }
+      message MetadataCredentialsFromPlugin {
+        string name = 1;
+        oneof config_type {
+          google.protobuf.Struct config = 2 [deprecated = true];
+          google.protobuf.Any typed_config = 3;
+        }
+      }
+      oneof credential_specifier {
+        string access_token = 1;
+        Empty google_compute_engine = 2;
+        string google_refresh_token = 3;
+        ServiceAccountJWTAccessCredentials service_account_jwt_access = 4;
+        GoogleIAMCredentials google_iam = 5;
+        MetadataCredentialsFromPlugin from_plugin = 6;
+      }
+    }
+    repeated CallCredentials call_credentials = 3;
+    string stat_prefix = 4;
+    string credentials_factory_name = 5;
+    google.protobuf.Struct config = 6;
+  }
+  oneof target_specifier {
+    EnvoyGrpc envoy_grpc = 1;
+    GoogleGrpc google_grpc = 2;
+  }
+  reserved 4;
```

Comment:
`google.protobuf.Duration timeout = 3;` is missing, compared to upstream.

Reply:
Added. @jplevyak The lite proto does not have this, it is to avoid having a copy of Duration proto?